R/GeospatialFunctions.R

```diff
@@ -450,28 +803,143 @@ TADA_GetATTAINS <- function(.data, return_sf = TRUE) {
           dplyr::distinct(assessmentunitidentifier, .keep_all = TRUE),
         silent = TRUE
       )
+      if(is.null(ATTAINS_polygons) || nrow(ATTAINS_polygons) == 0){
+        ATTAINS_polygons <- NULL
+      }
+      
     }))
-
+    # If there are ATTAINS catchments, fill_catchments = FALSE, and return_sf = TRUE:
     return(list(
       "TADA_with_ATTAINS" = TADA_with_ATTAINS,
       "ATTAINS_catchments" = ATTAINS_catchments,
       "ATTAINS_points" = ATTAINS_points,
       "ATTAINS_lines" = ATTAINS_lines,
       "ATTAINS_polygons" = ATTAINS_polygons
     ))
-  }
+    
+    # If there IS at least some ATTAINS data, and fill_catchments = TRUE...
+  } else if (!is.null(nearby_catchments) & fill_catchments == TRUE) {
+    # ... link WQP features to the ATTAINS catchment feature(s) they land in:
+    TADA_with_ATTAINS <- TADA_DataRetrieval_data %>%
+      sf::st_join(., nearby_catchments, left = TRUE)
+    
+    missing_attains <- dplyr::filter(TADA_with_ATTAINS, is.na(ATTAINS.submissionid))
+    
+    # Splitting up sites with and without ATTAINS, so remove those without ATTAINS:
+    TADA_with_ATTAINS <- TADA_with_ATTAINS %>%
+      dplyr::filter(!is.na(ATTAINS.submissionid))
+    
+    # If there are no WQP observations without missing ATTAINS features, return empty df for 
+    # TADA_without_ATTAINS
+    if(nrow(missing_attains) == 0){
+      
+      print('All WQP features are associated with an ATTAINS assessment unit. Returning empty dataframe for "TADA_without_ATTAINS".')
+      
+      if(resolution %in% c("NHDPlusV2", "NHDPlus V2", "MedRes", "LowRes", "NHDPlus")){
+        TADA_without_ATTAINS <- tibble::tibble(comid = character(),
+                                               resolution = character(),
+                                               catchmentareasqkm = numeric())
+        nhd_catchments <- NULL
+      } else if(resolution %in% c("NHDPlus HiRes", "HighRes", "NHDHR", "HR", "NHD HR")){
+        TADA_without_ATTAINS <- tibble::tibble(nhdplusid = character(),
+                                               resolution = character(),
+                                               catchmentareasqkm = numeric()) 
+        
+        nhd_catchments <- NULL
+      } else {stop('Please select between "NHDPlus V2" or "NHDPlus HiRes" for your NHD resolution.')}
```

Comment:
suggestion: Might be cleaner to have only two options available and to check at the top of the function and immediately stop if one of the two isn't detected. That way this can just be if(hi_res){...}else if(med_res){...}